src/Data/Aeson/V2.hs

```diff
@@ -0,0 +1,131 @@
+module Data.Aeson.V2
+  ( decode
+  , decode'
+  , eitherDecode
+  , eitherDecode'
+  , encode
+  , encodeFile
+  , decodeStrict
+  , decodeFileStrict
+  , decodeStrict'
+  , decodeFileStrict'
+  , eitherDecodeStrict
+  , eitherDecodeFileStrict
+  , eitherDecodeStrict'
+  , eitherDecodeFileStrict'
+  , Value (..)
```

Comment:
Re-exporting `Value`'s constructors seems to somewhat defeat the purpose of the compat package, since `Object`'s field is going to vary from `HashMap` to `Map` across aeson 1 and 2.